3_javascript/javascript1/week2/script.js

```diff
@@ -0,0 +1,96 @@
+console.log("Hey there!")
+
+//fullname
+function getFullname(firstname, surname) {
+  return `${firstname} ${surname}`;
+}
+
+const fullname1 = getFullname("Benjamin", "Hughes");
+const fullname2 = getFullname("John", "Smith");
+
+console.log(fullname1);
+console.log(fullname2); 
+
+
+function getFullname(firstname, surname, useFormalName, gender) {
+  if (useFormalName) {
+    if (gender === 'male') {
+      return `Lord ${firstname} ${surname}`;
+    } else if (gender === 'female') {
+      return `Lady ${firstname} ${surname}`;
+    }
+  } else {
+    return `${firstname} ${surname}`;
+  }
+}
+
+const formalMaleName = getFullname("John", "Smith", true, 'male'); 
+const formalFemaleName = getFullname("Jody", "Foster", true, 'female');
+
+console.log(formalFemaleName)
+
+//event
+function getEventWeekday(daysFromToday) {
+  const weekdays = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+
+
+  const currentDate = new Date;
+  const targetDate = new Date(currentDate);
+  targetDate.setDate(currentDate.getDate()) + daysFromToday;
+
+  const weekdayIndex = targetDate.getDay();
+
+  const eventWeekday = weekdays[weekdayIndex];
+
+  return eventWeekday;
+}
+
+console.log(getEventWeekday(1));
+
+//weather 
+function suggestClothing(temperature) {
+  if (temperature >= 26) {
+    return "Consider a tank top, shorts or a dress";
+  } else if (temperature >= 18) {
+    return "Consider jeans, t-shirt and a cardigan.";
+  } else if (temperature >= 10) {
+    return "Consider a sweater or a jacket.";
+  } else {
+    return "You should wear a heavy coat.";
+  }
+}
+
+let clothesToWear = suggestClothing(13);
+console.log(clothesToWear);
+
+
+
+//student manager
+const class07Students = [];
+
+function addStudentToClass(studentName) {
+
+  if (class07Students.length >= 6) {
+    console.log("Cannot add more students to class 07");
+  } else if (studentName === "") {
+    
+    console.log("Student name cannot be empty");
+  } else if (class07Students.includes(studentName)){
+    console.log(`Student ${studentName} is already in the class`);
+ } else if (studentName.toLowerCase() === "queen") {
```

Comment:
The problem is with this condition that if there are 7 students already in the list it will not get this far but will enter the first if condition: 
```
if (class07Students.length >= 6) {
    console.log("Cannot add more students to class 07");
  } 
```
To make sure that we check for the queen you can either move this up to be the first check. Or you can add it to the first check like this: 
```
//check if the list length is greater then 6 and it is not the queen.
if (class07Students.length >= 6 & studentName.toLowerCase() != "queen") {
    console.log("Cannot add more students to class 07");
  }
```
In order to catch these issues it is a good idea to add more than 7 students to the list and then the queen to test that it works.